lib/saluki-components/src/destinations/datadog_metrics/endpoint/endpoints.rs

```diff
@@ -0,0 +1,152 @@
+use std::{
+    collections::{HashMap, HashSet},
+    str::FromStr,
+};
+
+use http::Uri;
+use regex::Regex;
+use saluki_error::GenericError;
+use saluki_metadata;
+use serde::Deserialize;
+use snafu::{ResultExt, Snafu};
+use url::Url;
+
+use crate::destinations::datadog_metrics::DEFAULT_SITE;
+
+/// AdditionalEndpoints holds the endpoints and api keys parsed
+/// from the user configuration.
+#[derive(Clone, Debug, Default, Deserialize)]
+pub struct AdditionalEndpoints(HashMap<String, Vec<String>>);
+
+/// Error type for invalid endpoints.
+#[derive(Debug, Snafu)]
+#[snafu(context(suffix(false)))]
+pub enum EndpointError {
+    Parse { source: url::ParseError, domain: String },
+}
+
+#[derive(Default)]
+/// Holds the mapping between domains and their api keys.
```

Comment:
Two small nits, fixed as part of the suggestion:

- We want derives to be right above the struct, and doc comments to be on top.
- Capitalization/casing for common acronyms like `API`.

```suggestion
/// A single API endpoint and its associated API keys.
#[derive(Default)]
```